### ISSUE TITLE
pre / post backup hook

### ISSUE THREAD

**Issue by pgstef** [orgrim/pg_back#25]
Hi,

It often happens that we'd like to copy the dump generated to an external location. It would be nice to have pre/post backup hooks to do that rather than having to build a complete script around pg_back.

Would it be something possible ?

Kind regards

**Comment by orgrim** [comment]
Hello,

Sorry for the delay in may answer.

pre/post hook are interesting. For instance, pre hooks could make the code smaller by let us put the pausing of the replay on standby elsewhere.

As for post hook, I think they are interesting only if they are asynchronous. But this would be difficult to implement with bash.

Regards,

**Comment by jmrenouard** [comment]
Hi @pgstef 

Nice idea.

How to handle good cases from bad ones ?
For example, how to know that we can copy file on a remote location or not ?

Do we always run pre-hook even after failure ?

Br,

**Comment by jmrenouard** [comment]
Hi @pgstef @orgrim 

Pre backup script: (before starting pg_back himself)
   Always executed
   If return code is different from 0, pg_back fails and run directly Post backup script

Post backup script: (after pg_back execution)
  Always executed
  One parameter: global return code (pg_back + pre backup script if exists)
   If return code is different from 0, pg_back fails also.

So, if a problem occurs, pre and post script will be executed in all situations.
post script will be a garbage collector of error in pre script in all situations.
If pre script, post script or core pg_back fails return code couldn't be 0(SUCCESS)
